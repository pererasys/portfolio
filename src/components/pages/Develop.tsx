/**
 * @author Andrew Perera
 * Copyright (C) 2021
 */

import React, { ChangeEvent, useReducer } from "react";
import { Logo } from "../navigation/Header";
import { FormInput, TextBox } from "../common/Inputs";
import { Button } from "../common/Buttons";

import styles from "../../styles/pages/Develop.module.scss";

export const Header = () => (
  <header className={styles.header}>
    <Logo />
  </header>
);

export const PageRoot: React.FC = (props) => (
  <div className={styles.root}>{props.children}</div>
);

interface ILanding {
  title: string;
  subtitle: string;
}

export const Landing = (props: ILanding) => (
  <section className={styles.landing}>
    <h1>{props.title}</h1>
    <p className={styles.subtitle}>{props.subtitle}</p>
  </section>
);

const NAME_CHANGED = "name_changed";
const EMAIL_CHANGED = "email_changed";
const DESCRIPTION_CHANGED = "description_changed";

interface IAction {
  type: string;
  payload: string;
}

const initialFormState = {
  name: "",
  email: "",
  description: "",
};

function formReducer(state: typeof initialFormState, action: IAction) {
  switch (action.type) {
    case NAME_CHANGED:
      return { ...state, name: action.payload };
    case EMAIL_CHANGED:
      return { ...state, email: action.payload };
    case DESCRIPTION_CHANGED:
      return { ...state, description: action.payload };
    default:
      return state;
  }
}

export const Form = () => {
  const [form, dispatch] = useReducer(formReducer, initialFormState);

  const onNameChange = (event: ChangeEvent<HTMLInputElement>) => {
    dispatch({ type: NAME_CHANGED, payload: event.target.value });
  };

  const onEmailChange = (event: ChangeEvent<HTMLInputElement>) => {
    dispatch({ type: EMAIL_CHANGED, payload: event.target.value });
  };

  const onDescriptionChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
    dispatch({ type: EMAIL_CHANGED, payload: event.target.value });
  };

  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    console.log(form);
  };

  return (
    <form className={styles.form} onSubmit={onSubmit}>
      <FormInput
        id="email"
        name="email"
        type="email"
        label="Email Address"
        onChange={onEmailChange}
        required
      />
      <FormInput
        id="name"
        name="name"
        type="text"
        label="Full Name"
        onChange={onNameChange}
      />
      <TextBox
        id="description"
        name="description"
        label="Project Description"
        onChange={onDescriptionChange}
        required
      />
      <Button label="Submit" loadingLabel="Just a moment..." type="submit" />
    </form>
  );
};
