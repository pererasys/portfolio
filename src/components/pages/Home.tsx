/**
 * @author Andrew Perera
 * Copyright (C) 2021
 */

import Image from "next/image";
import Link from "next/link";

import { IconProp } from "@fortawesome/fontawesome-svg-core";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCode } from "@fortawesome/free-solid-svg-icons";
import { faBitcoin, faMedium } from "@fortawesome/free-brands-svg-icons";

import styles from "../../styles/pages/Home.module.scss";

interface IActivity {
  icon: IconProp;
  title: string;
  description: string;
  color: string;
}

const Activity = (props: IActivity) => (
  <li className={styles[`${props.color}Skill`]}>
    <FontAwesomeIcon icon={props.icon} className={styles.icon} />
    <h3>{props.title}</h3>
    <p>{props.description}</p>
  </li>
);

const ACTIVITIES = [
  {
    icon: faCode,
    title: "Software Developer",
    description:
      "Fullstack developer specializing in modern web technologies (React, GraphQL, AWS, etc.).",
    color: "purple",
  },
  {
    icon: faBitcoin,
    title: "Investor",
    description:
      "Active investor in equity and crypto markets, with a focus on deep value and high growth assets.",
    color: "darkBlue",
  },
  {
    icon: faMedium,
    title: "Author",
    description:
      "Growing following on platforms like Medium, with articles published in publications such as TheStartup.",
    color: "orange",
  },
];

export const Activities = () => (
  <section className={styles.activities}>
    <div className={styles.header}>
      <h1>Activities</h1>
      <p>Get to know me by exploring my professional interests.</p>
    </div>

    <ul>
      {ACTIVITIES.map((skill, index) => (
        <Activity {...skill} key={String(index)} />
      ))}
    </ul>
  </section>
);

export const Profile = () => (
  <section className={styles.profile}>
    <div>
      <Image
        src="/images/me.jpeg"
        height={300}
        width={300}
        layout="intrinsic"
        className={styles.profileImage}
      />
    </div>
    <div className={styles.content}>
      <h1>About Me</h1>
      <p>
        I'm a self-taught developer from Charleston, SC and recent graduate of
        University of South Carolina with a focus in Computer Science and
        Business.
      </p>
      <Link href="/about">
        <a className={styles.action}>See more</a>
      </Link>
    </div>
  </section>
);

export const LandingContent = () => <div></div>;
