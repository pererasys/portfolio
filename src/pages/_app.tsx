/**
 * @author Andrew Perera
 * Copyright (C) 2021
 */

import { AppProps } from "next/app";
import Router from "next/router";

import NProgress from "nprogress";

import { config } from "@fortawesome/fontawesome-svg-core";
import "@fortawesome/fontawesome-svg-core/styles.css";
config.autoAddCss = false;

import "../styles/globals.scss";
import "../styles/nprogress.scss";

Router.events.on("routeChangeStart", () => NProgress.start());
Router.events.on("routeChangeComplete", () => NProgress.done());
Router.events.on("routeChangeError", () => NProgress.done());

function App({ Component, pageProps }: AppProps) {
  return <Component {...pageProps} />;
}

export default App;
