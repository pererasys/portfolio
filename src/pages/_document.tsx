/**
 * @author Andrew Perera
 * Copyright (C) 2021
 */

import Document, {
  Head,
  Main,
  NextScript,
  DocumentContext,
  Html,
} from "next/document";

class CustomDocument extends Document {
  static async getInitialProps(ctx: DocumentContext) {
    const initialProps = await Document.getInitialProps(ctx);
    return { ...initialProps };
  }

  render() {
    return (
      <Html lang="en">
        <Head>
          <link rel="icon" href="/favicon.ico" />
          <link
            href="https://fonts.googleapis.com/css2?family=Open+Sans:wght@300;400;600&family=Rubik:wght@300;400;500;600;700;800&display=swap"
            rel="stylesheet"
          />
        </Head>
        <body id="body">
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}

export default CustomDocument;
