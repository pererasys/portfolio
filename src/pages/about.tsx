/**
 * @author Andrew Perera
 * Copyright (C) 2021
 */

import Page from "../components/navigation/Page";

import Landing from "../components/common/Landing";

export default function About() {
  return (
    <Page title="About" description="About me.">
      <Landing
        title="About"
        subtitle="Learn a little bit about me and what I do."
        position="left"
      />
    </Page>
  );
}
