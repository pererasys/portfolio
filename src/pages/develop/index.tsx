/**
 * @author Andrew Perera
 * Copyright (C) 2021
 */

import Page from "../../components/Page";

import {
  Form,
  Header,
  Landing,
  PageRoot,
} from "../../components/pages/Develop";

export default function Develop() {
  return (
    <Page
      title="Develop"
      description="Tell me about your project!"
      HeaderComponent={Header}
      RootComponent={PageRoot}
    >
      <Landing title="Develop" subtitle="Tell me about your project!" />
      <Form />
    </Page>
  );
}
