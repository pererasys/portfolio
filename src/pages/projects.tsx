/**
 * @author Andrew Perera
 * Copyright (C) 2021
 */

import Page from "../components/navigation/Page";

import Landing from "../components/common/Landing";

export default function Projects() {
  return (
    <Page title="Projects" description="A few of my projects.">
      <Landing
        title="Projects"
        subtitle="Explore my projects."
        position="left"
      />
    </Page>
  );
}
