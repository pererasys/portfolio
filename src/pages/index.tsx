/**
 * @author Andrew Perera
 * Copyright (C) 2021
 */

import Page from "../components/navigation/Page";

import Landing from "../components/common/Landing";
import { LandingContent, Profile, Activities } from "../components/pages/Home";

export default function Home() {
  return (
    <Page title="Home" description="description">
      <Landing title="Andrew Perera" subtitle="I build things." />
      <Profile />
      <Activities />
    </Page>
  );
}
