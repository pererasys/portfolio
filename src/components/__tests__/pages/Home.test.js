import { create } from "react-test-renderer";

import * as Home from "../../pages/Home";

describe("<Home.Profile />", () => {
  const setup = () => {
    const component = create(<Home.Profile />);

    return {
      component,
    };
  };

  it("should render without crashing", () => {
    const { component } = setup();

    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("<Home.Activities />", () => {
  const setup = () => {
    const component = create(<Home.Activities />);

    return {
      component,
    };
  };

  it("should render without crashing", () => {
    const { component } = setup();

    expect(component.toJSON()).toMatchSnapshot();
  });
});
