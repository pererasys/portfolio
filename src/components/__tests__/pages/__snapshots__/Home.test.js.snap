// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home.Activities /> should render without crashing 1`] = `
<section>
  <div>
    <h1>
      Activities
    </h1>
    <p>
      Get to know me by exploring my professional interests.
    </p>
  </div>
  <ul>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-code fa-w-20 "
        data-icon="code"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M278.9 511.5l-61-17.7c-6.4-1.8-10-8.5-8.2-14.9L346.2 8.7c1.8-6.4 8.5-10 14.9-8.2l61 17.7c6.4 1.8 10 8.5 8.2 14.9L293.8 503.3c-1.9 6.4-8.5 10.1-14.9 8.2zm-114-112.2l43.5-46.4c4.6-4.9 4.3-12.7-.8-17.2L117 256l90.6-79.7c5.1-4.5 5.5-12.3.8-17.2l-43.5-46.4c-4.5-4.8-12.1-5.1-17-.5L3.8 247.2c-5.1 4.7-5.1 12.8 0 17.5l144.1 135.1c4.9 4.6 12.5 4.4 17-.5zm327.2.6l144.1-135.1c5.1-4.7 5.1-12.8 0-17.5L492.1 112.1c-4.8-4.5-12.4-4.3-17 .5L431.6 159c-4.6 4.9-4.3 12.7.8 17.2L523 256l-90.6 79.7c-5.1 4.5-5.5 12.3-.8 17.2l43.5 46.4c4.5 4.9 12.1 5.1 17 .6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <h3>
        Software Developer
      </h3>
      <p>
        Fullstack developer specializing in modern web technologies (React, GraphQL, AWS, etc.).
      </p>
    </li>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bitcoin fa-w-16 "
        data-icon="bitcoin"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zm-141.651-35.33c4.937-32.999-20.191-50.739-54.55-62.573l11.146-44.702-27.213-6.781-10.851 43.524c-7.154-1.783-14.502-3.464-21.803-5.13l10.929-43.81-27.198-6.781-11.153 44.686c-5.922-1.349-11.735-2.682-17.377-4.084l.031-.14-37.53-9.37-7.239 29.062s20.191 4.627 19.765 4.913c11.022 2.751 13.014 10.044 12.68 15.825l-12.696 50.925c.76.194 1.744.473 2.829.907-.907-.225-1.876-.473-2.876-.713l-17.796 71.338c-1.349 3.348-4.767 8.37-12.471 6.464.271.395-19.78-4.937-19.78-4.937l-13.51 31.147 35.414 8.827c6.588 1.651 13.045 3.379 19.4 5.006l-11.262 45.213 27.182 6.781 11.153-44.733a1038.209 1038.209 0 0 0 21.687 5.627l-11.115 44.523 27.213 6.781 11.262-45.128c46.404 8.781 81.299 5.239 95.986-36.727 11.836-33.79-.589-53.281-25.004-65.991 17.78-4.098 31.174-15.792 34.747-39.949zm-62.177 87.179c-8.41 33.79-65.308 15.523-83.755 10.943l14.944-59.899c18.446 4.603 77.6 13.717 68.811 48.956zm8.417-87.667c-7.673 30.736-55.031 15.12-70.393 11.292l13.548-54.327c15.363 3.828 64.836 10.973 56.845 43.035z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <h3>
        Investor
      </h3>
      <p>
        Active investor in equity and crypto markets, with a focus on deep value and high growth assets.
      </p>
    </li>
    <li>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-medium fa-w-14 "
        data-icon="medium"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 32v448h448V32H0zm372.2 106.1l-24 23c-2.1 1.6-3.1 4.2-2.7 6.7v169.3c-.4 2.6.6 5.2 2.7 6.7l23.5 23v5.1h-118V367l24.3-23.6c2.4-2.4 2.4-3.1 2.4-6.7V199.8l-67.6 171.6h-9.1L125 199.8v115c-.7 4.8 1 9.7 4.4 13.2l31.6 38.3v5.1H71.2v-5.1l31.6-38.3c3.4-3.5 4.9-8.4 4.1-13.2v-133c.4-3.7-1-7.3-3.8-9.8L75 138.1V133h87.3l67.4 148L289 133.1h83.2v5z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <h3>
        Author
      </h3>
      <p>
        Growing following on platforms like Medium, with articles published in publications such as TheStartup.
      </p>
    </li>
  </ul>
</section>
`;

exports[`<Home.Profile /> should render without crashing 1`] = `
<section>
  <div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <noscript>
        <img
          decoding="async"
          src="/images/me.jpeg"
          srcSet="/_next/image?url=%2Fimages%2Fme.jpeg&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fme.jpeg&w=640&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
      <img
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div>
    <h1>
      About Me
    </h1>
    <p>
      I'm a self-taught developer from Charleston, SC and recent graduate of University of South Carolina with a focus in Computer Science and Business.
    </p>
    <a
      href="/about"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      See more
    </a>
  </div>
</section>
`;
